tests/test_hmgnn.py

```
@@ -0,0 +1,521 @@
+        batch_size=BATCH_SIZE_HMGNN,
+    )
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="CUDA not available or PyTorch CUDA setup issue")
+class TestCrossScaleAttention:
+    def test_instantiation(self):
+        attn_scale_dims = [HIDDEN_DIM_HMGNN] * NUM_SCALES
+        num_scales_val = len(attn_scale_dims)
+        # CROSS_ATTN_HIDDEN_DIM is the dimension to which inputs are projected and used internally.
+        attention = CrossScaleAttentionMH(n_scales=num_scales_val, hidden_dim=CROSS_ATTN_HIDDEN_DIM)
+        assert attention.S == num_scales_val  # Check .S attribute which stores n_scales
```

Ruff (0.11.9)

Yoda condition detected

Rewrite as `num_scales_val == attention.S`

(SIM300)